readme.md

```diff
@@ -53,7 +53,76 @@ Directory names should be lowercase. Filenames should be singular.
+
+To do this:
+
+**Step 1** (only need to do this once)
+
+Create a local clone of `wikimedia-page-library`:
+```
+git clone https://github.com/wikimedia/wikimedia-page-library.git
+```
+
+**Step 2** (do this as needed)
+
+Checkout the branch to be tested:
+```
+cd ~/wikimedia-page-library/
+git checkout somebranch
+```
+
+**Step 3** (only need to do this once)
+
+Now we need to temporarily tell NPM to use our local copy of `wikimedia-page-library` so when we run the app's build process and it runs `npm install`, our local copy of  `wikimedia-page-library` will be used instead of the published one.
+
+We can use the `npm link` command to do this.
+
+Create a `wikimedia-page-library` symlink:
+```
+cd ~/wikimedia-page-library/
+npm link
+```
+
+The `npm link` command above configures NPM with a global `wikimedia-page-library` symlink. Note that it infers the name of the package from the package folder we `cd`'ed into. We will use this symlink in the next step. You only have to create this symlink once as long as you don't change the file system location of your local clone of `wikimedia-page-library`.
+
+**Step 4** (do this when you begin testing against a local copy)
+
+Now we have to configure the app's `www/node_modules/` folder to point to the symlink we just created.
+```
+`cd` to the app's `www/node_modules/` folder
+npm link wikimedia-page-library
+```
+
+After running the `link` command above, your file system browser should show a symlink folder for `www/node_modules/wikimedia-page-library` instead of a normal folder. 
+
+Now NPM is "pointing" to our local copy of `wikimedia-page-library` and when the app's build process invokes `npm install` NPM will try to get `wikimedia-page-library` build products from our local copy. 
+
+**Step 5** (do this each time you make a change to the local copy)
+
+*Important!* 
+
+Don't forget to rebuild the local copy of `wikimedia-page-library` any time you make a change to it (before you run the app's build process), otherwise the app's build process won't "see" the change!
+```
+cd ~/wikimedia-page-library/
+npm run-script build
+```
+
+You may find it helpful to combine this step's command with a command kicking of the respective app's build process into a shortcut of some kind. Most of the development workflow involves make a change to your local copy of `wikimedia-page-library`, telling that local copy to re-build, then telling the app to rebuild.
+
+**Step 6** (do this when you are done testing against the local copy)
+
+When you have finished testing against your local copy of `wikimedia-page-library`, you can simply `unlink` so the app's build process will again pull from the *published* version of `wikimedia-page-library`.
+```
+`cd` to the app's `www/node_modules/` folder
```

Comment:
It seems odd to use code snippet notation inside of a code block. Consider something like:

```sh
cd native-app-repo/www # Replace with the path to the Android or iOS repo.
npm unlink wikimedia-page-library
```

Or:

```sh
cd $NATIVE_APP_REPO/www # Replace NATIVE_APP_REPO with the path to the Android or iOS repo.
npm unlink wikimedia-page-library
```